dotnet/src/Functions/Functions.Prompty/Extensions/PromptyKernelExtension.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.IO;
+using Azure.AI.OpenAI;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.Connectors.OpenAI;
+using Microsoft.SemanticKernel.PromptTemplates.Liquid;
+using Microsoft.SemanticKernel.Prompty.Core;
+using YamlDotNet.Serialization;
+
+namespace Microsoft.SemanticKernel.Prompty.Extension;
+
+/// <summary>
+/// extension methods for <see cref="Kernel"/> to create a <see cref="KernelFunction"/> from a prompty file.
+/// </summary>
+public static class PromptyKernelExtension
+{
+    /// <summary>
+    /// Create a <see cref="KernelFunction"/> from a prompty file.
+    /// </summary>
+    /// <param name="_">kernel</param>
+    /// <param name="promptyPath">path to prompty file.</param>
+    /// <param name="promptTemplateFactory">prompty template factory, if not provided, a <see cref="LiquidPromptTemplateFactory"/> will be used.</param>
+    /// <param name="loggerFactory">logger factory</param>
+    /// <returns><see cref="KernelFunction"/></returns>
+    /// <exception cref="ArgumentNullException"></exception>
+    /// <exception cref="NotSupportedException"></exception>
+    public static KernelFunction CreateFunctionFromPrompty(
+        this Kernel _,
+        string promptyPath,
+        IPromptTemplateFactory? promptTemplateFactory = null,
+        ILoggerFactory? loggerFactory = null)
+    {
+        var text = File.ReadAllText(promptyPath);
+
+        promptTemplateFactory ??= new LiquidPromptTemplateFactory(); // use liquid template factory by default
```

Comment:
You can use `AggregatorPromptTemplateFactory` to combine multiples formats if you want

Reply:
Cool, seems that handlerbar template format can also be easily support